src/kubernetes.cc

```diff
@@ -787,12 +893,12 @@ void KubernetesReader::WatchPods(
   const std::string pod_label_selector(
       config_.KubernetesPodLabelSelector().empty()
       ? "" : "&" + config_.KubernetesPodLabelSelector());
+  const std::string api_path =
+      WatchPath("pods", "v1", node_selector + pod_label_selector);
 
   try {
     WatchMaster(
```

Comment:
Would it be possible to also end up with a single `WatchObjects` that can be adapted to pods and nodes via parameters?

Reply:
Passing in the callback properly is the tricky thing - let me give this a go.